projects/.metadata/.plugins/org.eclipse.core.resources/.history/25/00a1e777f21d00181353d905c66b77c4

```diff
@@ -0,0 +1,384 @@
+
+#include <sys/platform.h>
+#include "ADuCM3029_demo_cn0398.h"
+#include <common.h>
+#include <common/adi_timestamp.h>
+#include "Timer.h"
+
+using namespace adi_sensor_swpack;
+
+/* Static data */
+static ADI_DATA_PACKET    			gSensorData;
+static ADI_STRING_DATA				gCommData = {
+													.nStringSize = 13u};
+static ADI_WIFI_TCP_CONNECT_CONFIG  gTCPConnect = {
+													.nLinkID = ADI_WIFI_CONNECTION_ID,
+													.pType = (uint8_t *) "TCP",
+													.pIP   = aMQTTBrokerIp,
+													.pPort =  aMQTTBrokerPort,
+													.nTCPKeepAlive = ADI_WIFI_MQTT_PUBLISHER_KEEPALIVE};
+
+static ADI_WIFI_MQTT_CONNECT_CONFIG gMQTTConnect = {
+													.nLinkID = ADI_WIFI_CONNECTION_ID,
+													.nVersion = ADI_WIFI_MQTT_PUBLISHER_VERSION,
+													.pName = aMQTTPublisherName,
+													.nMQTTKeepAlive = ADI_WIFI_MQTT_PUBLISHER_KEEPALIVE };
+
+static ADI_WIFI_PUBLISH_CONFIG 		gPublishConfig = {
+		 	 	 	 	 	 	 	 	 	 	 	 .pMQTTData = (uint8_t *) &gSensorData,
+		 	 	 	 	 	 	 	 	 	 	 	 .nMQTTDataSize = sizeof(ADI_DATA_PACKET),
+													 .pTopic = aMQTTTopicName,
+													 .nLinkID = ADI_WIFI_CONNECTION_ID,
+													 .nQos = ADI_WIFI_MQTT_PUBLISER_QOS};
+
+uint8_t m_spi_memory[ADI_SPI_MEMORY_SIZE];
+
+/* Local Wi-Fi functions */
+static void            InitWiFiConnection(void);
+static void            adi_wifi_AplicationCallback(void * pCBParam, uint32_t nEvent, void * pArg);
+static void			   MQTTPublishCN0398(CN0398 pCn);
+
+/* Local system functions */
+static void InitSystem(void);
+static void Trap(void);
+static void RepairConnection(void);
+
+int main(int argc, char *argv[])
+{
+    CN0398         	cn0398;
+    SENSOR_RESULT   eSensorResult;
+
+	/**
+	 * Initialize managed drivers and/or services that have been added to 
+	 * the project.
+	 * @return zero on success 
+	 */
+	
+    /* Initialize the system */
+    InitSystem();
+
+    /* Open ADC */
+    eSensorResult = cn0398.open();
+
+    if(eSensorResult != SENSOR_ERROR_NONE) {
+        PRINT_SENSOR_ERROR(DEBUG_MESSAGE, eSensorResult);
+        Trap();
+    }
+
+    /* Start measurement */
+    eSensorResult = cn0398.start();
+
+    if(eSensorResult != SENSOR_ERROR_NONE) {
+        PRINT_SENSOR_ERROR(DEBUG_MESSAGE, eSensorResult);
+        Trap();
+    }
+
+	/* Publish AD7124 ADC data using MQTT */
+	MQTTPublishCN0398(cn0398);
+
+	return 0;
+}
+
+/*!
+ * @brief      Application Callback
+ *
+ * @details    Called by the framework layer (adi_wifi_noos.c) when an event occurs.
+ *
+ * @param [in] pCBParam : Callback parameter (unused)
+ *
+ * @param [in] Event : Event of type #ADI_WIFI_AT_CMDCODE.
+ *
+ * @param [in] pArg : Callback argument (unused)
+ *
+ */
+void adi_wifi_AplicationCallback(void * pCBParam, uint32_t nEvent, void * pArg)
+{
+	switch (nEvent) {
+			case CMD_NONE:
+				/* Read the publish Data. */
+				adi_wifi_ParseSubscriberData();
+				break;
+			case CMD_HW_ERROR:
+			    break;
+			default:
+				break;
+		}
+
+	return;
+}
```

Comment:
case should be under the switch, you have a few extra tabs, it should be something like
```
switch (action) {
case KOBJ_ADD:
        return "add";
case KOBJ_REMOVE:
        return "remove";
case KOBJ_CHANGE:
        return "change";
default:
        return NULL;
}
```

take a look at: https://www.kernel.org/doc/html/v4.10/process/coding-style.html#placing-braces-and-spaces